eos-paygd/main.c

```diff
@@ -240,6 +295,9 @@ main (int   argc,
   GOptionContext *context;
   gboolean enforcing_mode = TRUE;
 
+  log_io = g_io_channel_new_file (LOG_FILE_NAME, "a", NULL);
```

Comment:
Is the real `/var` accessible when `eos-payg` starts up?

You are not handling failure from this function, indicated by it returning `NULL`. https://docs.gtk.org/glib/ctor.IOChannel.new_file.html

You could write:

```
  log_io = g_io_channel_new_file (LOG_FILE_NAME, "a", &error);
  if (log_io != NULL)
    {
      g_log_set_writer_func (log_writer, log_io, NULL);
    }
  else
    {
      g_warning ("Failed to open %s: %s", LOG_FILE_NAME, error->message);
      g_clear_error (&error);
    }
```

Reply:
> Is the real `/var` accessible when `eos-payg` starts up?
> 

Hum, you are right, we need to wait until pivoting to the real root, which means we'll lose some startup messages (I). That is fine IMO, as these will still be in the journal and are either fatal errors or redundant debug info: nothing that would affect an analysis of credit consumption.